code/components/citizen-server-impl/src/NoticeLogicProcessor.cpp

```diff
@@ -0,0 +1,149 @@
+#include <StdInc.h>
+#include "NoticeLogicProcessor.h"
+
+#include <ResourceManager.h>
+
+NoticeLogicProcessor::NoticeLogicProcessor(fx::ServerInstanceBase* server)
+{
+	m_cvManager = server->GetComponent<console::Context>()->GetVariableManager();
+
+	server->GetComponent<fx::ResourceManager>()->ForAllResources([this](const fwRefContainer<fx::Resource>& resource)
+	{
+		if (resource->GetState() == fx::ResourceState::Started)
+		{
+			m_resourceList.push_back(resource->GetName());
+		}
+	});
```

Comment:
Can you put this into `fx::ResourceManager` like discussed?

Reply:
Toyed around with it a bit and ended up ditching the local cache & instead chose to tap into ResourceManager's GetResource within the logic processor loop.

Implementing a limited use resource identity/name cache into the manager itself didn't really sit well with me. If we chose to go that route we'd need to outline when such cache should be refreshed or what its lifetime is supposed to be in general, which feels a bit overkill at the moment.

Even without this cache vector evaluation is still very fast - the latest commit includes this 😄

Reply:
Your solution is what I had more or less in mind. My idea was to just let it find the resource and return its status or another status if it didn't exist. But making this processor agnostic to the resource system was my main goal, which looks a lot better now. Getting rid of these caches was my second goal, the use of those are generally not needed.

So good job!